### PR TITLE
dockerclient: implement COPY --chmod

### LINKED ISSUE #228: dockerclient quietly ignores the --chmod flag for ADD/COPY
We extended the parser to recognize it and add a field to `Copy` structs, but didn't teach `dockerclient` to either use the value or produce an error when people attempted to use the flag.

### PR DESCRIPTION
**Pull request by nalind**
We've been recognizing the argument for some time, we just ignored it.  Fixes #228.

### CODE REVIEW

File: dockerclient/client.go [hunk 1]
```diff
@@ -1093,6 +1106,13 @@ func (e *ClientExecutor) CopyContainer(container *docker.Container, excludes []s
 				}
 				r = filtered
 			}
+			if c.Chmod != "" {
```
Comment (rhatdan) [inline dockerclient/client.go:1109]: Should this only happen if c.Chown != "" {?

Comment (nalind) [reply dockerclient/client.go:1109]: Oof, no, it should not.

Comment (nalind) [reply dockerclient/client.go:1109]: Wait, it doesn't...?

Comment (rhatdan) [reply dockerclient/client.go:1109]: Your right the way github is displaying it confused me.

LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**